apiserver/resources.go

```diff
@@ -144,19 +150,18 @@ func (h *ResourcesHandler) readResource(backend ResourcesBackend, req *http.Requ
 	var res resource.Resource
 	if uReq.PendingID != "" {
 		res, err = backend.GetPendingResource(uReq.Application, uReq.Name, uReq.PendingID)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
 	} else {
 		res, err = backend.GetResource(uReq.Application, uReq.Name)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
+	}
+	if err != nil {
+		return nil, errors.Trace(err)
 	}
 
-	ext := path.Ext(res.Path)
-	if path.Ext(uReq.Filename) != ext {
-		return nil, errors.Errorf("incorrect extension on resource upload %q, expected %q", uReq.Filename, ext)
+	if res.Type == charmresource.TypeFile {
```

Comment:
For docker resources we should validate that the content parses as a valid docker image url

Reply:
Should call checkDockerDetails() for docker resources here